x/valset-pref/validator_set.go

```diff
@@ -46,6 +54,7 @@ func (k Keeper) GetValidatorSetPreference(ctx sdk.Context, delegator string) (ty
 
 // SetValidatorSetPreference creates or updates delegators validator set.
 // Errors when the given preference is the same as the existing preference in state.
+// NOTE: this function doesnot add valset to the state
 func (k Keeper) SetValidatorSetPreference(ctx sdk.Context, delegator string, preferences []types.ValidatorPreference) (types.ValidatorSetPreferences, error) {
```

Comment:
Why is it called Set? Lets rename to derive?

Reply:
changed: https://github.com/osmosis-labs/osmosis/pull/5967/commits/90e9fb2fd747e95310c0b8c26bc518af76fe366e#diff-5d02bf1a8724ea9309dc52d49cfa6d9effb872a27c7e0757d48a016a10f9d42aR29